### ISSUE TITLE
replaceIdentifiers damages source map mappings for deconflicted identifiers

### ISSUE THREAD

**Issue by krisselden** [rollup/rollup#101]
This makes it impossible to remap back to the original source modules after minification, because the names are all misaligned.

example
```
Warning: mismatched names
Expected: FRAGMENT_SHADER || 'FRAGMENT_SHADER' || 'FRAGMENT_SHADER' || "FRAGMENT_SHADER" || "FRAGMENT_SHADER"
Got:     return this ||     return this.c ||     return this.c ||     return this.c ||     return this.c
Original Line:     return this.compileShader(FRAGMENT_SHADER, source);
Mapping: 1:22935→52:0 "FRAGMENT_SHADER" in ../lib/rendering_context.js
```

from https://github.com/krisselden/webgl-classes/blob/master/bin/build.js

**Comment by eventualbuddha** [comment]
Hey @krisselden :smile: Perhaps @Rich-Harris knows what you're talking about here, but I admit to being a bit lost. Do you have a reproducible example I can work with?

**Comment by krisselden** [comment]
@eventualbuddha I'll try to make a simpler example but things seem to get messed up by the naming deconflict (which should also create name entries in the source map).

**Comment by krisselden** [comment]
@eventualbuddha ok, here is the simple example this was created with `rollup index.js -m inline -o bundle.js`

http://goo.gl/DlRN9E

**Comment by krisselden** [comment]
I also checked in that example, with reproduction steps and what to look for.

https://github.com/krisselden/test_rollup_sourcemap

It is somewhere in this code https://github.com/rollup/rollup/blob/master/src/Statement.js#L286

**Comment by krisselden** [comment]
Even if the mappings for the replaced name get fixed, it really should be adding the name to the names section of the source map

**Comment by Rich-Harris** [comment]
Thanks for flagging this up @krisselden, it's the kick in the arse I needed to fix this issue upstream. I've just released magic-string 0.7.0 with the relevant changes. Thanks also for the test case, it's very helpful

**Comment by Rich-Harris** [comment]
Fixed in 0.15.0

**Comment by krisselden** [comment]
@Rich-Harris awesome, thank you!
